internal/services/containerapps/container_app_resource.go

```diff
@@ -0,0 +1,414 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: managedenvironments.ValidateManagedEnvironmentID,
+			Description:  "The ID of the Container App Environment to host this Container App.",
+		},
+
+		"template": helpers.ContainerTemplateSchema(),
+
+		"revision_mode": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(containerapps.ActiveRevisionsModeSingle),
+				string(containerapps.ActiveRevisionsModeMultiple),
+			}, true),
+			DiffSuppressFunc: suppress.CaseDifference,
+		},
+
+		"ingress": helpers.ContainerAppIngressSchema(),
+
+		"registry": helpers.ContainerAppRegistrySchema(),
+
+		"secret": helpers.SecretsSchema(),
+
+		"dapr": helpers.ContainerDaprSchema(),
+
+		// "identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ContainerAppResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationComputed(),
+
+		"outbound_ip_addresses": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"latest_revision_name": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The name of the latest Container Revision.",
+		},
+
+		"latest_revision_fqdn": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Description: "The FQDN of the Latest Revision of the Container App.",
+		},
+
+		"custom_domain_verification_id": {
+			Type:        pluginsdk.TypeString,
+			Computed:    true,
+			Sensitive:   true,
+			Description: "The Custom Domain Verification ID for the Container App.",
+		},
+	}
+}
+
+func (r ContainerAppResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ContainerApps.ContainerAppClient
+			environmentClient := metadata.Client.ContainerApps.ManagedEnvironmentClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var app ContainerAppModel
+
+			if err := metadata.Decode(&app); err != nil {
+				return err
+			}
+
+			id := containerapps.NewContainerAppID(subscriptionId, app.ResourceGroup, app.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			envId, err := managedenvironments.ParseManagedEnvironmentID(app.ManagedEnvironmentId)
+			if err != nil {
+				return fmt.Errorf("parsing Container App Environment ID for %s: %+v", id, err)
+			}
+
+			env, err := environmentClient.Get(ctx, *envId)
+			if err != nil || env.Model == nil {
```

Comment:
we shouldn't need to check this since the SDK will raise an error if not?

```suggestion
			if err != nil {
```